osquery/main/shell.cpp

```diff
@@ -83,6 +89,43 @@ int profile(int argc, char *argv[]) {
   return 0;
 }
 
+// readline completion expects strings to be malloced. readline will free them
+// later.
+char *copy_string(const std::string &str) {
+  char *copy = (char *)malloc(str.size() + 1);
+  if (copy == nullptr) {
+    LOG(ERROR)
```

Comment:
On second thought, are we sure this logging does not need new heap data? The app may be SOL at this point.